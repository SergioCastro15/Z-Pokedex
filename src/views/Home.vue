<template>
  <div class="container">
    <poke-input
      @triggerSearchPokemon="getPhotos"
    />
      <h1 class="mt-5">{{getCurrentPokemon.name}}</h1>
      <b-button
        class="my-4"
        @click="getPokemon"
        block
        variant="success"
      >
        Get a random Pokemon
      </b-button>
    <poke-card
      class="mt-5"
      :pokemon="getCurrentPokemon.sprites"
    />
    <poke-abilities class="my-5" :abilities="getCurrentPokemon.abilities" />
  </div>
</template>

<script>
import PokeCard from '../components/PokeCard.vue';
import PokeInput from '../components/PokeInput.vue';
import PokeAbilities from '../components/PokeAbilities.vue';

export default {
  name: 'Home',
  components: {
    PokeCard,
    PokeInput,
    PokeAbilities,
  },
  beforeMount() {
    this.getPokemon();
  },
  computed: {
    getCurrentPokemon() {
      return this.$store.state.pokemon;
    },
  },
  methods: {
    getRandomPokemon() {
      return Math.round(Math.random() * 150);
    },
    searchPokemon(text) {
      this.$store.dispatch('getPokemon', text);
    },
    getPokemon() {
      this.$store.dispatch('getPokemon', this.getRandomPokemon());
    },
    getPhotos() {
      this.$store.dispatch('getPhotos');
    },
  },
};
</script>
