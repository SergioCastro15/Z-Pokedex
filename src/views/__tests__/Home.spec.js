import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Home from '../Home.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Home component', () => {
  let actions;
  let store;
  let state;

  beforeEach(() => {
    state = {
      pokemon: {},
    };

    actions = {
      getPokemon: jest.fn(),
    };

    store = new Vuex.Store({
      state,
      actions,
    });
  });

  it('render', () => {
    const wrapper = shallowMount(Home, {
      store: {
        ...store,
        state: {
          ...state,
          pokemon: {},
        },
      },
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should return current component', () => {
    const localThis = {
      $store: {
        state: {
          pokemon: 'pikachu',
        },
      },
    };
    const response = Home.computed.getCurrentPokemon.call(localThis);
    expect(response).toBe('pikachu');
  });

  it('should run searchPokemon method', () => {
    const localThis = {
      $store: {
        dispatch(method, payload) {
          expect(method).toBe('getPokemon');
          expect(typeof payload).toBe('string');
        },
      },
    };
    Home.methods.searchPokemon.call(localThis, 'pikachu');
  });

  it('should run getPokemon method', () => {
    const localThis = {
      getRandomPokemon() {
        return 3;
      },
      $store: {
        dispatch(method, payload = 3) {
          expect(method).toBe('getPokemon');
          expect(payload).toBe(localThis.getRandomPokemon());
        },
      },
    };
    Home.methods.getPokemon.call(localThis);
  });
});
