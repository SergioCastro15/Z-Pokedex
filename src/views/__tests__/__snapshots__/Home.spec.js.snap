// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component render 1`] = `
<div
  class="container"
>
  <poke-input-stub />
   
  <h1
    class="mt-5"
  >
    
  </h1>
   
  <b-button
    block=""
    class="my-4"
    variant="success"
  >
    
      Get a random Pokemon
    
  </b-button>
   
  <poke-card-stub
    class="mt-5"
  />
   
  <poke-abilities-stub
    class="my-5"
  />
</div>
`;
